### PR TITLE
add POLYGON metadata

### PR DESCRIPTION
**Pull request by lcxywfe**
Append to #18614.

Changes proposed in this pull request:
- support POLYGON metadata
Just followed #18649

### CODE REVIEW

**Comment by lcxywfe** [comment]
@zjcnb Please help review, thx.

**Comment by zjcnb** [comment]
> @zjcnb Please help review, thx.

The CI has error, fix it please

File: shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/dialect/MySQLSchemaMetaDataLoaderTest.java [hunk 1]
```diff
@@ -88,12 +88,12 @@ private ResultSet mockTypeInfoResultSet() throws SQLException {
...
     private ResultSet mockTableMetaDataResultSet() throws SQLException {
         ResultSet result = mock(ResultSet.class);
-        when(result.next()).thenReturn(true, true, true, true, true, true, false);
+        when(result.next()).thenReturn(true, true, true, true, true, true, true, false);
         when(result.getString("TABLE_NAME")).thenReturn("tbl");
-        when(result.getString("COLUMN_NAME")).thenReturn("id", "name", "doc", "geo", "t_year", "mp");
-        when(result.getString("DATA_TYPE")).thenReturn("int", "varchar", "json", "geometry", "year", "multipolygon");
-        when(result.getString("COLUMN_KEY")).thenReturn("PRI", "", "", "", "", "");
-        when(result.getString("EXTRA")).thenReturn("auto_increment", "", "", "", "", "");
+        when(result.getString("COLUMN_NAME")).thenReturn("id", "name", "doc", "geo", "t_year", "pg", "mpg");
```
Comment (mabaiwan) [inline shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/dialect/MySQLSchemaMetaDataLoaderTest.java:93]: @lcxywfe Why change from mg to mpg?

Comment (lcxywfe) [reply shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/dialect/MySQLSchemaMetaDataLoaderTest.java:93]: For aligning to ```pg```

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/dialect/MySQLSchemaMetaDataLoader.java [hunk 2]
```diff
@@ -126,6 +126,7 @@ private void appendDataTypes(final Map<String, Integer> dataTypes) {
         dataTypes.putIfAbsent("GEOMETRY", Types.BINARY);
         dataTypes.putIfAbsent("YEAR", Types.DATE);
         dataTypes.putIfAbsent("MULTIPOINT", -2);
+        dataTypes.putIfAbsent("POLYGON", -2);
```
[the remark this answers is not in the document]
Comment (lcxywfe) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/database/schema/loader/dialect/MySQLSchemaMetaDataLoader.java:129]: DONE

**Pull request merged**